src/Interpreters/FillingRow.cpp

```diff
@@ -16,6 +16,10 @@ bool less(const Field & lhs, const Field & rhs, int direction)
 
 bool equals(const Field & lhs, const Field & rhs)
 {
+    /// This will treat NaNs as equal
+    if (lhs.getType() == rhs.getType())
```

Comment:
should we rather extend `FieldVisitorAccurateEquals` and use `FloatCompareHelper` if both values are floats?

Reply:
I checked the code, and this was my concern as well. I thought that FieldVisitorAccurateEquals tries to compare exactly as the underlying values compare. But I'm not sure if it is needed. So - simple change first...